### PR TITLE
salt/utils/cloud: Allow to customize ssh gateway command/options

### PR DESCRIPTION
**Pull request by icy**
### What does this PR do?

By default, nc command (`nc -q0 %h %p`) is used to create gateway. This is not
always possible in some restricted environments. An alternative is to use
native support from ssh: `-W %h:%p`.

This commit allows user to provide a customize option for gateway. Please note
the (wait_for_port) process may need another patch.

### Previous Behavior

Users are forced to use `nc -q0 %h %p` command to create ssh proxy for `salt-cloud`

### New Behavior

Users can use a customized command to create ssh proxy for `salt-cloud`

### Tests written?

NO

### Commits signed with GPG?

Yes

### CODE REVIEW

File: doc/man/salt.7 [hunk 1]
```diff
@@ -80063,6 +80063,10 @@ my\-ec2\-config:
   # Optional
   ssh_gateway_username: root
 
+  # Default to nc -q0 %h %p
+  # Optional
+  ssh_gateway_command: "-W %h:%p"
+
```
Comment (gtmanfred) [inline doc/man/salt.7]: Can you remove this change, we compile these every time we build, so this will not remain.

**Review by gtmanfred — CHANGES_REQUESTED** [review]
This should be documented in salt/cloud/clouds/ec2.py, as well as adding the gateway stuff to ./doc/topics/cloud/misc.rst

File: doc/man/salt.7 (hunk 1 above)
Comment (icy) [reply doc/man/salt.7]: thanks a lot. I will update `ec2.py` and `misc.rst` instead :)

**Review by gtmanfred — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by icy** [comment]
Thanks @gtmanfred and @cachedout  for your reviews and approvals.

**Comment by rallytime** [comment]
Thanks @icy and congrats on your first salt PR. :)

**Comment by icy** [comment]
Hi @rallytime and @gtmanfred,

Is there any good start to write and improve unit tests (e.g., in my PRs)?

At my laptop, the tests are passed as seen here https://gist.github.com/icy/a59f975ba54224b7d941b2e38ccc74a1 , I don't think these tests are working fine. I did some smoke tests actually and it's not a recommended way (expensive?)

**Pull request merged**